Test/Unit/Gateway/Request/RefundRequestTest.php

```diff
@@ -0,0 +1,134 @@
+use Magento\Payment\Model\Method\Logger;
+use Magento\Sales\Api\Data\OrderPaymentInterface;
+use Magento\Sales\Model\Order\Payment;
+
+class RefundRequestTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @var Logger
+     */
+    private $logger;
+    /**
+     * @var Data
+     */
+    private $helper;
+
+    protected function setUp(): void
+    {
+        $this->configMock = $this->createMock(ConfigInterface::class);
+        $this->orderMock = $this->createMock(OrderAdapterInterface::class);
+        $this->paymentDO = $this->createMock(PaymentDataObjectInterface::class);
+        $this->paymentModel = $this->getMockBuilder(Payment::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        
+        $this->helper = $this->getMockBuilder(Data::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->logger = $this->getMockBuilder(Logger::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->refundRequest = new RefundRequest($this->logger, $this->helper);
+    }
+
+    public function testBuild()
+    {
+        $amount = "10.00";
+        $command = "cc:voidrefund";
+        $refnum = '23443535';
+        
+        $buildSubject = [
+            'payment' => $this->paymentDO,
+            'amount' => $amount
+        ];
+
+        $expectation = [
+            'xCommand' => $command,
+            'xAmount'   => $this->helper->formatPrice($amount),
+            'xRefNum' => null,
+        ];
+        $this->paymentDO->expects(static::once())
+            ->method('getPayment')
+            ->willReturn($this->paymentModel);
+      
+        $this->paymentDO->expects(static::once())
+            ->method('getOrder')
+            ->willReturn($this->orderMock);
+
+        static::assertEquals(
+            $expectation,
+            $this->refundRequest->build($buildSubject)
+        );
+    }
+
+    public function testBuildRefund()
+    {
+        $amount = "10.00";
+        $command = "cc:refund";
+        $refnum = '23443535';
+        $expectation = [
+            'xCommand' => $command,
+            'xAmount'   => $this->helper->formatPrice($amount),
+            'xRefNum' => null,
+        ];
+
+        $buildSubject = [
+            'payment' => $this->paymentDO,
+            'amount' => $amount
+        ];
+
+        $this->paymentDO->expects(static::once())
+            ->method('getPayment')
+            ->willReturn($this->paymentModel);
+
+        $this->paymentDO->expects(static::once())
+            ->method('getOrder')
+            ->willReturn($this->orderMock);
+
+        $this->orderMock->expects(static::any())
+            ->method('getGrandTotalAmount')
+            ->willReturn($amount);
+
+        static::assertEquals(
+            $expectation,
+            $this->refundRequest->build($buildSubject)
+        );
+        // $this->refundRequest->build($buildSubject);
```

Comment:
Is this comment needed?

Reply:
No, it's not needed. I have removed that comment code and pushed it.